packages/dds/sequence/src/sequence.ts

```diff
@@ -656,6 +656,7 @@ export abstract class SharedSegmentSequence<T extends MergeTree.ISegment>
             if (error) {
                 this.logger.sendErrorEvent({ eventName: "SequenceLoadFailed" }, error);
                 this.loadedDeferred.reject(error);
```

Comment:
should we return this.loadedDeferred.promise at the bottom of the function instead?

Reply:
there is a non-async usage of this function, returning the promise would make the function async. It technically wouldn't cause a problem, as that case is a non-error case, but i think i could introduce bugs if someone took advantage of the async-ness to do other things in the load finished method